pandas/tests/groupby/test_groupby.py

```diff
@@ -1009,6 +1009,36 @@ def test_groupby_complex():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "frame,expected",
+    [
+        (
+            DataFrame(
+                [
+                    {"a": 1, "b": 1 + 1j},
+                    {"a": 1, "b": 1 + 2j},
+                    {"a": 4, "b": 1},
+                ]
+            ),
+            DataFrame(
+                np.array([1, 1, 1], dtype=np.int64),
+                index=Index([(1 + 1j), (1 + 2j), (1 + 0j)], dtype="object", name="b"),
+                columns=Index(["a"], dtype="object"),
+            ),
+        )
+    ],
+)
```

Comment:
if theres just 1 case, better to not parametrize (clearer, plus smaller memory footprint)

Reply:
done!